src/ophyd_async/core/_device.py

```diff
@@ -106,12 +108,15 @@ async def connect(
             Time to wait before failing with a TimeoutError.
         """
         can_use_previous_connect = (
-            mock is self._connect_mock_arg
+            bool(mock) is self._connect_mock_arg
             and self._connect_task
             and not (self._connect_task.done() and self._connect_task.exception())
         )
+        if mock is True:
+            mock = Mock()  # create a new Mock if one not provided
+        self.mock = mock
```

Comment:
Hmm, having seen this I'm inclined to go back to having a global to store all the mocks and MockSignalBackends in, like we had ages ago:
https://github.com/bluesky/ophyd-async/blob/54deda30cae717472d1020bd5a67489380960ad8/src/ophyd_async/core/signal.py#L24

Reply:
Okay, do we want the mapping to be

- dict[Signal, MockSignalBackend]
- dict[MockSignalBackend, Mock]
- dict[Signal, Mock]

Reply:
```python
_mock_signal_backends: dict[Signal, MockSignalBackend]  # used for `set_mock_value`
_device_mocks: dict[Device, Mock]  # used for `get_mock`
```